From 8aa8f7cf9f5be53b4ffe916a47c3915d9baf8a99 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Tue, 13 Sep 2011 01:06:10 -0300
Subject: [PATCH 063/154] pvr: Unwrap pvr memory after use

Memory that has been wrapped should be unwrapped after use
---
 sys/pvr2d/gstpvrvideosink.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index cf8acfc..b4fb8b6 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -319,6 +319,13 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
       break;
   }
   dcontext->wsegl_table->pfnWSEGL_SwapDrawable (dcontext->drawable_handle, 1);
+  PVR2DQueryBlitsComplete (dcontext->pvr_context, dcontext->dst_mem, 1);
+  pvr_error = PVR2DMemFree (dcontext->pvr_context, dcontext->src_mem);
+  if (pvr_error != PVR2D_OK) {
+    GST_ERROR_OBJECT (pvrvideosink, "Failed to Unwrap buffer memory"
+        "returned %d", pvr_error);
+    return;
+  }
   GST_DEBUG_OBJECT (pvrvideosink, "end");
 }
 
-- 
1.7.5.4

